core/spec/models/spree/promotion_handler/shipping_spec.rb

```diff
@@ -13,8 +13,20 @@ module PromotionHandler
       context 'with apply_automatically' do
         let!(:promotion) { create(:promotion, apply_automatically: true, promotion_actions: [action]) }
 
-        it "creates the adjustment" do
-          expect { subject.activate }.to change { shipment.adjustments.count }.by(1)
+        context 'for eligible promotion' do
+          it "creates the adjustment" do
+            expect { subject.activate }.to change { shipment.adjustments.count }.by(1)
+          end
+        end
+
+        context 'for ineligible promotion' do
+          let(:promotion) do
+            create(:promotion, :with_item_total_rule, item_total_threshold_amount: 1_000, code: 'freeshipping', promotion_actions: [action])
```

Comment:
You want to set `apply_automatically: true` in this context rather than `code: 'freeshipping'`, right?

Reply:
My bad, copied that as I wasn't familiar with the factories. Also needs to be a `let!`.